slapd-cli

```diff
@@ -227,6 +227,17 @@ else
 	message "info" "[INFO] Using built-in configuration - this may cause some problems"
 fi
 
+# Use systemd to manage the service if it's present
+_use_systemctl=0
+if [ $PPID -ne 1 -a -z "$SYSTEMCTL_SKIP_REDIRECT" ] && \
+    [ -d /run/systemd/system ] && \
```

Comment:
Not sure if  -d /run/systemd/system is a sufficient test, for example when slapd is run into docker.
See for example: https://github.com/ltb-project/openldap-deb/commit/b1c55b585e156b670fa5f85bfb5251f2dcf2d41c

```
  if ! type "systemctl" > /dev/null 2>&1; then
    # No SYSTEMD available
    # do not start on boot
    # do not start slapd process now
    echo && exit $?
  else
```

Reply:
In a docker container, /run/systemd does not exist because systemd is not running in the container. In that case, slapd-cli will not redirect to systemd.

Testing whether systemctl exists or not is not a good test, because a user could have systemctl installed, but NOT be using systemd as their init system (corner case, maybe)

[ -d /run/systemd/system] is the test distros already do to see whether or not systemd is PID1 or not. See /lib/lsb/init-functions.d/40-systemd in debian and /etc/init.d/functions on redhat. So I am pretty confident about this being the right test to check whether systemd is currently managing the systemd or not.